libraries/chain/include/eosio/chain/pending_snapshot.hpp

```diff
@@ -33,9 +33,15 @@ class pending_snapshot {
       return snapshots_dir / fc::format_string(".incomplete-snapshot-${id}.bin", fc::mutable_variant_object()("id", block_id));
    }
 
-   T finalize(const chain::controller& chain) const {
+   T finalize(const block_id_type& lib_id, const chain::controller& chain) const {
+      auto lib_num = chain::block_header::num_from_id(lib_id);
+      auto block_num = chain::block_header::num_from_id(block_id);
+
+      assert(lib_num >= block_num);
+
+      // finalize called before forkdb is pruned of non-irreversible blocks, so this can find a non-irreversible block
       auto block_ptr = chain.fetch_block_by_id(block_id);
```

Comment:
imo it'd be nice to remove `block_ptr` since it and the new comment above are now red herrings on how this works -- the check for `block_ptr` for `in_chain` doesn't really have anything to do with _whether it's in chain_, it's just there to guard against bad usage of it later on. A more useful comment would be why `lib_num > block_num` is sufficient for being considered in chain. (what we discussed here about snapshots immediately at start)

But it's not easy to remove due to `block_ptr->timestamp` below.

Reply:
I changed it to store timestamp so it doesn't need to be retrieved. Also improved the comments.

Reply:
lgtm :shipit: